sound/soc/sof/intel/mtl.c

```diff
@@ -110,6 +110,15 @@ static int mtl_ipc_send_msg(struct snd_sof_dev *sdev, struct snd_sof_ipc_msg *ms
 	snd_sof_dsp_write(sdev, HDA_DSP_BAR, MTL_DSP_REG_HFIPCXIDR,
 			  msg_data->primary | MTL_DSP_REG_HFIPCXIDR_BUSY);
 
+	/* No need to schedule d0i3 work when sending PM messages */
+	if (SOF_IPC4_MSG_IS_MODULE_MSG(msg_data->primary) &&
+	    (SOF_IPC4_MSG_TYPE_GET(msg_data->primary) == SOF_IPC4_MOD_SET_DX ||
+	    SOF_IPC4_MSG_TYPE_GET(msg_data->primary) == SOF_IPC4_MOD_SET_D0IX))
+		return 0;
+
+	mod_delayed_work(system_wq, &hdev->d0i3_work,
+			 msecs_to_jiffies(SOF_HDA_D0I3_WORK_DELAY_MS));
+
```

Comment:
this should be a common helper, repeating the code 3 times isn't so good.

Reply:
sure, thanks!